### PR TITLE
refactor(project): improve loading components and error logic

### PR DESCRIPTION
**Pull request by BlackHole1**

### CODE REVIEW

File: desktop/renderer-app/src/pages/PeriodicRoomDetailPage/index.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const PeriodicRoomDetailPage = observer<{}>(function PeriodicRoomDetailPa
     }, [periodicUUID, roomStore]);
 
     if (!periodicInfo || !rooms) {
-        return <LoadingPage />;
+        return <LoadingPage timeMS={3 * 1000} />;
```
Comment (Cyberhan123) [inline desktop/renderer-app/src/pages/PeriodicRoomDetailPage/index.tsx:45]: please use global variables

**Comment by BlackHole1** [comment]
In the current example, the use of variables may not work particularly well.

Generally magic number variable is more applicable to places where there is no obvious intention.

And here, already with `Loading`, `time` and `MS`, I think that's enough information.

@Cyberhan123

File: desktop/renderer-app/src/pages/PeriodicRoomDetailPage/index.tsx (hunk 1 above)
Comment (Cyberhan123) [inline desktop/renderer-app/src/pages/PeriodicRoomDetailPage/index.tsx:45]: please leave some comments. It will take some time to understand

**Comment by Cyberhan123** [comment]
> In the current example, the use of variables may not work particularly well.
> 
> Generally magic number variable is more applicable to places where there is no obvious intention.
> 
> And here, already with `Loading`, `time` and `MS`, I think that's enough information.
> 
> @Cyberhan123

Ok, we can use option type. also can easy reset value. I may keep my opinion
```ts
export interface LoadingPageProps {
    text?: string;
    time?: 'default' | 'faster';
}
```

File: desktop/renderer-app/src/pages/PeriodicRoomDetailPage/index.tsx (hunk 1 above)
Comment (BlackHole1) [reply desktop/renderer-app/src/pages/PeriodicRoomDetailPage/index.tsx:45]: yes

**Review by Cyberhan123 — APPROVED** [review]

**Comment by BlackHole1** [comment]
> > In the current example, the use of variables may not work particularly well.
> > Generally magic number variable is more applicable to places where there is no obvious intention.
> > And here, already with `Loading`, `time` and `MS`, I think that's enough information.
> > @Cyberhan123
> 
> Ok, we can use option type. also can easy reset value. I just keep my opinion
> 
> ```ts
> export interface LoadingPageProps {
>     text?: string;
>     time?: 'default' | 'faster';
> }
> ```

I'm wondering if this approach, hiding the exact time directly, would not be good.

cc: @netless-io/developers

**Pull request merged**